pkg/apis/certificates/types.go

```diff
@@ -43,6 +43,10 @@ type CertificateSigningRequestSpec struct {
 	// Base64-encoded PKCS#10 CSR data
 	Request []byte `json:"request"`
 
+	// signingProfile specifies a requested signing profile for a CA to use
+	// when issuing the certificate.
+	// SigningProfile string `json:"signingProfile,omitempty"`
```

Comment:
The intent is to provide the same as storage class to persistent volume claim. It allows an administrator to pick specific permutations of configuration to requestors without exposing to full configuration matrix, where many of the configurations are invalid.

> not sure we want to let all csr clients view signingprofiles

Why?